From e4a77de591dc562d06ed5bd9ad7c2ed3eada4f80 Mon Sep 17 00:00:00 2001
From: Rubin Xu <rubinxu@google.com>
Date: Wed, 15 Sep 2021 21:43:18 +0800
Subject: [PATCH] DO NOT MERGE KeyChain protect against overlays.

Set setFilterTouchesWhenObsucred on positive confirmation button.

Test: N/A
Bug: 199754277
Change-Id: I096afafb74b69e86e82f89f6121ba52d65c99564
(cherry picked from commit 90c6182ef5933ec69383115e2703a177a0c5929b)
---
 src/com/android/keychain/KeyChainActivity.java | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/com/android/keychain/KeyChainActivity.java b/src/com/android/keychain/KeyChainActivity.java
index 1279ab2..7165a09 100644
--- a/src/com/android/keychain/KeyChainActivity.java
+++ b/src/com/android/keychain/KeyChainActivity.java
@@ -438,6 +438,9 @@ public class KeyChainActivity extends Activity {
                 finish(null);
             }
         });
+        dialog.create();
+        // Prevents screen overlay attack.
+        dialog.getButton(DialogInterface.BUTTON_POSITIVE).setFilterTouchesWhenObscured(true);
         dialog.show();
     }
 
-- 
2.34.0.rc2.393.gf8c9666880-goog

